### PR TITLE
Feature/test module routes - almost done

### PR DESCRIPTION
**Pull request by Alsje15**
Fixed the timeout error of running the google strategy twice with before (mocha) hook.
Added tests for all the module routes we have now.

The addModule and removeModule tests pass but don't work;
I don't think I'm implementing them right but I've tried many different things and am hoping you see something I don't.
Anyway, will come back to them after some refactoring - for now they're clean and don't break anything :)

### CODE REVIEW

File: app/test/module.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          done();
+        });
+      });
+    });
+
+    describe("POST /module/addModule", () => {
+      it("should add a Module from Module Table to respective user (UserModule table)", (done) => {
+        agent
+          .post("/module/addModule")
+          .send(mod)
```
[the remark this answers is not in the document]
Comment (mellifluus) [reply app/test/module.js:126]: p.s. - seems to be working for me but you should be checking if it works for you too

File: app/test/module.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          .end((err, res) => {
+            expect(err).to.be.null;
+            expect(res).to.have.status(200);
+            expect(res.body).to.be.a("object");
+            done();
+          });
+      });
+    });
```
Comment (mellifluus) [inline app/test/module.js:134]: this test should be ran before the `getAllUserModule` test. This way, you first test it with an empty db, and expect length 0. Then, after adding one userModule, you can test it for a response of length 1.

File: app/test/module.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    describe("POST /module/removeModule", () => {
+      it("should remove respective Module and User from UserModule table", (done) => {
+        agent
+          .post("/module/removeModule")
+          .send(mod)
```
Comment (mellifluus) [inline app/test/module.js:140]: same `...` operator here

**Review by mellifluus — COMMENTED** [review]
this is all i can find. with these fixes, all tests pass for me. Also remember to run `npm run setup:test` before running `npm run test` every time

**Comment by mellifluus** [comment]
I also suggest you change the script `"test": "cross-env NODE_ENV=test mocha ./app/test"` in the `package.json` file to `"test": "npm run setup:test && cross-env NODE_ENV=test mocha ./app/test"`. This way, every time you run tests with `npm run test` it's going to first setup the db and then run the tests

**Comment by Alsje15** [comment]
> this is all i can find. with these fixes, all tests pass for me. Also remember to run `npm run setup:test` before running `npm run test` every time

Thank you! Yes, I'm not using setup:test anymore but instead testing our seeder data after setup:db because we will always know what's in it and then actually have dummy data to work with x

File: app/test/module.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alsje15) [reply app/test/module.js:126]: Life saver, thank yoouu :D

**Pull request merged**